[aydrienlaw] iP

aydrienlaw opened the pull request.

aydrienlaw committed:
Reformat: Rename Duke class to Tilo

aydrienlaw committed:
Update logo

aydrienlaw committed:
.gitignore: Add *.class

aydrienlaw committed:
Tilo: Add sayHi, sayGoodbye, printBorder methods

aydrienlaw committed:
Add echo feature: implement readUserInput and echo methods

aydrienlaw committed:
Add task management feature: implement addTask and listTasks methods

aydrienlaw committed:
Update task management feature: add Task class and mark/unmarkTask methods

aydrienlaw committed:
Update Tilo class: refactor task management methods and enhance task representation

ycdaniel326 commented on src/main/java/Tilo.java:
```diff
@@ -0,0 +1,112 @@
+
+public class Tilo {
+    public static final int taskListSize = 100;
+    public static final String INDENT = "\t";
```
I like the idea of defining the Indent and border line here! 
And perhaps you can consider changing taskListSize to upper case if it is a constant.

aydrienlaw committed:
Add todo feature: ToDo class and addToDo, handleAddTaskCommand methods

ycdaniel326 commented on src/main/java/Tilo.java:
```diff
@@ -0,0 +1,112 @@
+            userInput = readUserInput(in);
+            printBorder();
+            String[] words = userInput.split(" ");
+            switch (words[0]) {
```
I like how you use switch statements to separate different cases. 
For the format, it seems that we don't need to add indentation for the case clauses.

ycdaniel326 commented on src/main/java/Tilo.java:
```diff
@@ -0,0 +1,112 @@
+        System.out.println(INDENT + "added: " + line);
+        tasks[taskCount] = new Task(line);
+    }
+
```
I like how you write separate methods here to make your main function shorter and readable.

aydrienlaw committed:
Add deadline feature: Deadline class and addDeadline method

aydrienlaw committed:
Add event feature: Event class and addEvent method

aydrienlaw committed:
Refactor Tilo class: encapsulate fields, update methods to instance level

aydrienlaw committed:
Update input.txt and EXPECTED.TXT for text UI testing

seahsongli commented on src/main/java/Tilo.java:
```diff
@@ -3,96 +3,120 @@
+        Task newTask = new Task(description);
+        tasks.add(newTask);
+        printTaskAdded(newTask);
     }
```
I like the encapsulation implementation here. Should everything here be private?

seahsongli commented on src/main/java/Tilo.java:
```diff
@@ -3,96 +3,120 @@
-        String description = inputLine.substring(0, fromIndex);
-        String from = inputLine.substring(fromIndex + fromPrefix.length(), toIndex);
-        String to = inputLine.substring(toIndex + toPrefix.length());
+        if (fromIndex == -1 || toIndex == -1) {
+            System.out.println(INDENT + "Event format should be: event <description> /from <start> /to <end>");
+            return;
+        }
+
+        String description = inputLine.substring(0, fromIndex).trim();
+        String from = inputLine.substring(fromIndex + fromPrefix.length(), toIndex).trim();
+        String to = inputLine.substring(toIndex + toPrefix.length()).trim();
+
+        if (description.isEmpty() || from.isEmpty() || to.isEmpty()) {
+            System.out.println(INDENT + "Description, start time, and end time are all required.");
+            return;
+        }
+
```
I like the idea of error check here. Should the error check be in an enum instead of just a numerical value?

seahsongli commented on text-ui-test/EXPECTED.TXT:
I like the update of UI Testing case

aydrienlaw committed:
Organize files into packages

aydrienlaw committed:
Merge branch 'branch-A-Packages'

aydrienlaw committed:
Add TiloException class for custom exception handling

aydrienlaw committed:
Add Ui class for user interface handling

aydrienlaw committed:
Refactor Tilo class: delegate UI interactions to Ui class

aydrienlaw committed:
Add TaskList class for task management

aydrienlaw committed:
Refactor Tilo class: replace ArrayList with TaskList for task management

adoorknob commented on src/main/java/Tilo.java:
```diff
@@ -0,0 +1,220 @@
+    private static final int TASK_LIST_SIZE = 100;
+    private static final String INDENT = "\t";
+    private static final String BORDER = "----------------------------------------";
+    private static final String LOGO = INDENT + "___________.___.____     ________\n"
+            + INDENT + "\\__    ___/|   |    |   \\_____  \\\n"
+            + INDENT + "   |    |  |   |    |    /   |   \\\n"
+            + INDENT + "   |    |  |   |    |___/    |    \\\n"
+            + INDENT + "   |____|  |___|_______ \\_______  /\n"
+            + INDENT + "                       \\/       \\/";
```
you could wrap this in """ instead to keep the white spaces

adoorknob commented on src/main/java/Tilo.java:
```diff
@@ -0,0 +1,220 @@
+        return false;
+    }
+
+    private void printTaskAdded(Task task) {
```
good job having a separate method to print messages! could consider making a whole separate class as well

adoorknob commented on src/main/java/Tilo.java:
```diff
@@ -0,0 +1,220 @@
+
+    private void unmarkTask(Task task) {
+        task.markAsNotDone();
+        System.out.println(INDENT + "OK, I've marked this task as not done yet:");
```
could avoid using magic strings, would be easier to edit down the road as well

adoorknob commented on src/main/java/Tilo.java:
```diff
@@ -0,0 +1,220 @@
+        String[] words = userInput.split(" ", 2);
+        String command = words[0].toLowerCase();
+
+        switch (command) {
+        case "todo":
+            addToDo(words[1]);
+            break;
+        case "deadline":
+            addDeadline(words[1]);
+            break;
+        case "event":
+            addEvent(words[1]);
+            break;
+        case "list":
+            listTasks();
+            break;
+        case "mark":
+            handleMarkCommand(words, true);
+            break;
+        case "unmark":
+            handleMarkCommand(words,false);
+            break;
+        case "bye":
+            isRunning = false;
+            break;
+        default:
+            addTask(userInput);
+            break;
```
i like this switch case, very readable

adoorknob commented on src/main/java/Tilo.java:
```diff
@@ -0,0 +1,220 @@
+        printTaskAdded(newEvent);
+    }
+
+    private boolean isInputValid(String[] words) {
```
this method is named `isInputValid` but returns true if input is invalid

adoorknob reviewed (commented):
🎊 could consider having separate classes for printing messages and creating task types, otherwise good code quality overall! LGTM

aydrienlaw committed:
Add Command pattern and Parser class for user input processing

- Implement Command interface with execute method
- Add specific Command classes (AddToDoCommand, AddDeadlineCommand, etc.)
- Create Parser class to handle input parsing and validation
- Extract command creation logic from main Tilo class

aydrienlaw committed:
TiloException: Add new exception cases

aydrienlaw committed:
Command classes: Refactor to improve exception handling

aydrienlaw committed:
Merge branch 'branch-Level-5'

aydrienlaw committed:
refactor (TaskList): convert List<Task> to ArrayList<Task>

aydrienlaw committed:
feat (DeleteCommand): implement task deletion functionality

aydrienlaw committed:
refactor (TaskList): move class from tilo.task to tilo.storage

aydrienlaw committed:
refactor (Command classes): update taskList import statements

aydrienlaw committed:
feat (Storage): implement load and save methods for task management

aydrienlaw committed:
refactor (Command classes): rename parameters for clarity

aydrienlaw committed:
feat (Storage): implement loading and saving of tasks in Tilo class

aydrienlaw committed:
Merge branch 'branch-Level-6'

aydrienlaw committed:
Merge branch 'branch-Level-7'

aydrienlaw committed:
refactor(Tilo): extract methods from run()

- Extract processNextCommand() to handle single command execution
- Extract shutdown() to handle cleanup and goodbye logic

aydrienlaw committed:
refactor(Ui): improve code quality with constants and validation

aydrienlaw committed:
feat(TaskList): add markTask and unmarkTask methods

Extract task marking logic from command layer for better
separation of concerns

aydrienlaw committed:
refactor(Storage): enhance parsing robustness and error handling

- Extract constants for file format fields and validation
- Improve parsing with field count validation and error recovery

aydrienlaw committed:
feat(FindCommand): add keyword-based task search

Implemented keyword search for tasks.

Updated classes:
- Ui: display search results
- Parser: handle `find` command parsing
- TaskList: add findTasks method to filter tasks by keyword
- Task: add getDescription method
- FindCommand: execute search logic

aydrienlaw committed:
Merge pull request #1 from aydrienlaw/branch-Level-9

feat(find): add keyword-based task search

aydrienlaw committed:
refactor: improve code quality, error handling, and documentation

Major improvements to Parser, Storage, TaskList, Commands, TiloException, and Ui
with better validation, constants extraction, and proper layer separation.

aydrienlaw committed:
docs: add comprehensive JavaDoc comments to all classes

aydrienlaw committed:
Merge pull request #2 from aydrienlaw/branch-A-JavaDoc

docs: add comprehensive JavaDoc documentation to all classes

aydrienlaw committed:
docs: create user guide for all Tilo commands

aydrienlaw committed:
Merge pull request #3 from aydrienlaw/branch-A-UserGuide

docs: create user guide for all Tilo commands

aydrienlaw committed:
chore(TiloException): correct typo in corruptedLine error message
